src/shogun/mathematics/RandomMixin.h

```diff
@@ -0,0 +1,154 @@
+#ifndef __RANDOMMIXIN_H__
+#define __RANDOMMIXIN_H__
+
+#include <shogun/base/SGObject.h>
+#include <shogun/lib/config.h>
+
+#include <iterator>
+#include <random>
+#include <type_traits>
+
+namespace shogun
+{
+	namespace random
+	{
+		/** Seeds an SGObject using a random number generator as a seed source
+		 */
+		template <
+		    typename T, typename PRNG,
+		    std::enable_if_t<std::is_base_of<
+		        CSGObject, typename std::remove_pointer<T>::type>::value>* =
+		        nullptr>
+		static inline void seed(T* object, PRNG& prng)
```

Comment:
i'm really not so sure to get it why is this good.
if an SGO is a RandomMixin, then
 * the default ctored object should have a random seed (and prng seeded with that seed).
 * if the user wants to set a seed then call sgo.put("seed", 1)